### PR TITLE
Add command 'keys derive' to derive public key from private key

### PR DESCRIPTION
**Pull request by bluesign**
Closes https://github.com/onflow/flow-cli/issues/505

______

For contributor use:

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: internal/keys/generate.go [hunk 1]
```diff
@@ -30,8 +30,9 @@ import (
 )
 
 type flagsGenerate struct {
-	Seed       string `flag:"seed" info:"Deterministic seed phrase"`
-	KeySigAlgo string `default:"ECDSA_P256" flag:"sig-algo" info:"Signature algorithm"`
+	Seed           string `flag:"seed" info:"Deterministic seed phrase"`
+	FromPrivateKey string `flag:"from-private-key" info:"Use user private key instead of random"`
```
Comment (sideninja) [inline internal/keys/generate.go:34]: It would be nice to also support `FromPublicKey` which would get private key from public key :D just kidding

**Comment by sideninja** [comment]
Thank you, this is great. I was just thinking if it makes sense for this to be under the `flow keys generate` command as it's not really generating a key, but deriving a public key. Maybe a command `flow keys derive <private-key>` could be better.

**Comment by bluesign** [comment]
Sure I will update the PR, it makes more sense like that

**Review by sideninja — COMMENTED** [review]
🙏 One last thing to add would be documentation for this. You can just copy this page and make some changes to it https://github.com/onflow/flow-cli/blob/master/docs/generate-keys.md

File: docs/derive-keys.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+- Flag: `--sig-algo`
+- Valid inputs: `"ECDSA_P256", "ECDSA_secp256k1"`
+
+Specify the ECDSA signature algorithm for the key pair.
+
+Flow supports the secp256k1 and P-256 curves.
+
+
+### From File
```
Comment (sideninja) [inline docs/derive-keys.md:48]:
```suggestion
### Filter
```

**Review by sideninja — APPROVED** [review]
Thank you! Small comment, but LGTM

**Pull request merged**